ctools/defragment_sharded_collection.py

```diff
@@ -489,18 +489,21 @@ def __len__(self):
             merge_consecutive_chunks_without_size_check = False
 
             
-            def will_overflow_target_size():
+            def will_overflow_merge_cap():
                 """Returns true if merging the `consecutive_chunks` with the current one `c` will
-                produce a chunk that is 20% bigger that the target chunk size.
+                produce a chunk that is 20% bigger that the target chunk size or if we have reached
+                the user configured max number of chunks per batch.
 
                 If we don't trust the estimation of `consecutive_chunks` or 
                 we don't know the size of `c` this function will always return false.
                 """
+                if args.merge_batch_size is not None and len(consecutive_chunks) == args.merge_batch_size:
```

Comment:
[nit][nit][nit] In Python it's OK to say that
```suggestion
                if args.merge_batch_size and len(consecutive_chunks) == args.merge_batch_size:
```